WeatherParser.h

```diff
@@ -0,0 +1,87 @@
+#ifndef WEATHERPARSER_H
+#define WEATHERPARSER_H
+
+#include <qbytearray.h>
+#include <qjsondocument.h>
+#include <qjsonobject.h>
+#include <qjsonarray.h>
+#include <QDebug>
+#include <iostream>
+#include <boost/lexical_cast.hpp>
+
+#include "WeatherData.h"
+
+using namespace std;
+
+const static double gCALVIN = 273.15;
+
+class WeatherParser {
+private:
+    QJsonObject jsonRoot;
+
+    QJsonValue getGenericValueForKey (QString keyName, QJsonObject jsonObject){
+        return jsonObject.value(keyName);
+    }
+
+    std::string valueToStdString (QJsonValue value){
+        return value.toString().toStdString();
+    }
+
+    double calvinToCelsius(double value){
+        return value - gCALVIN;
+    }
+
+public:
+    WeatherParser(QByteArray jsonData) {
+        QJsonDocument sourceJsonDoc;
+        QJsonDocument destJsonDoc = sourceJsonDoc.fromJson(jsonData);
+        jsonRoot = destJsonDoc.object();
+    }
+
+    QJsonValue getValueForKey(QString keyName) {
+        return getGenericValueForKey(keyName, jsonRoot);
+        //return jsonRoot.value(keyName);
+    }
+
+    QJsonObject getObjectForKey(QString keyName) {
+        return getGenericValueForKey(keyName, jsonRoot).toObject();
+        //return jsonRoot.value(keyName).toObject();
+    }
+
+    QJsonArray getArrayForKey(QString keyName) {
+        return getGenericValueForKey(keyName, jsonRoot).toArray();
+        //return jsonRoot.value(keyName).toArray();
+    }
+
+    std::string getTemp(QJsonObject parent) {
+        QJsonValue jsonValue = getGenericValueForKey("temp", parent);
+        //QJsonValue jsonValue = parent.value(QString("temp"));
+        return boost::lexical_cast<std::string>(calvinToCelsius(jsonValue.toDouble()));
```

Comment:
Could you try:
std::string varAsString = std::to_string(myDoubleVar);
For convert double to string!?